### PR TITLE
Graphql experimental

### PR DESCRIPTION
**Pull request by micorix**
Added SearchNode so we can fully switch to GraphQL without losing any features.

Why creating new node? So we can extend it later with suggestions (when no school matches search).

### CODE REVIEW

File: warsawlo/schema.py [hunk 1]
```diff
@@ -116,6 +117,54 @@ class Meta:
...
+    # TODO: Filter input as list not string
+    # or filtering
+    available_extended_subjects = CharInFilter(
+        field_name="highschoolclass__extendedsubject__name"
+    )
+
+    def query_filter(self, queryset, value):
+        return (
+            queryset.annotate(similarity=TrigramSimilarity(self.QUERY_FIELD, value))
+            .filter(similarity__gte=0.05)
```
Comment (annkamsk) [inline warsawlo/schema.py:148]: In the future, I suggest checking also for words that contain the searched substring, cause if the school_name is a very long string (for example full name of Batory high school) and we're searching only for "bat", the value of trigram similarity is going to be small while we still expect to get the Batory high school. But this is for another PR.

File: warsawlo/schema.py [hunk 2]
```diff
@@ -116,6 +117,54 @@ class Meta:
...
+        model = School
+        fields = ["school_type", "is_public"]
+
+    query = django_filters.CharFilter(method="query_filter")
+
+    district = django_filters.CharFilter(field_name="address__district")
+
+    # TODO: Filter input as list not string
+    # or filtering
+    available_extended_subjects = CharInFilter(
```
Comment (annkamsk) [inline warsawlo/schema.py:141]: This should be probably MultipleChoiceFilter?

**Review by annkamsk — APPROVED** [review]

File: warsawlo/schema.py (hunk 2 above)
Comment (micorix) [reply warsawlo/schema.py:141]: Done

File: warsawlo/schema.py (hunk 1 above)
Comment (micorix) [reply warsawlo/schema.py:148]: Yeah, for now I'm leaving it as it is.

**Comment by micorix** [comment]
Closing it. I messed up git stuff again. Also we wanna merge it to master and use it on prod so it won't be "experimental" anymore.

**Pull request closed without merging**